syntax = "proto3";
package oneflow;

message Section {
  int32 first_id = 1;
  int32 last_id = 2;
}

enum ParallelPolicy {
  kDataParallel = 0;
  kModelParallel = 1;
}

message ParallelConf {
  ParallelPolicy policy = 1;
  repeated Section devices = 2;
}

message PlacementGroup {
  string name = 1;
  repeated string op_names = 2;
  ParallelConf parallel_conf = 3;
}

message Strategy {
  repeated PlacementGroup placement_groups = 1;
}
